server/services/db-manager.js

```diff
@@ -1,42 +1,48 @@
 const fs = require("fs");
 const pokemonClient = require("../clients/pokemon-client.js");
+const { Task } = require("../db/models");
 const POKEMON_FILE_NAME = "./server/pokemon-names.json";
-const TASKS_FILE_NAME = "./server/tasks.json";
 
-class FileManager {
-   getAllTasks() {
-      if (fs.existsSync(TASKS_FILE_NAME)) {
-         try {
-            const data = fs.readFileSync(TASKS_FILE_NAME);
-            const fileTasks = JSON.parse(data);
+class DBManager {
+   getAllTasks = async () => await Task.findAll();
 
-            return fileTasks;
-         } catch (error) {
-            error.statusCode = 500;
-            error.message = "Failed to read file";
-
-            throw error;
-         }
-      } else {
-         this._writeToFile(TASKS_FILE_NAME, []);
-
-         return [];
-      }
+   async addTask(task) {
+      await Task.bulkCreate(task);
+      await Task.sync();
    }
 
-   addTask(task) {
-      let allTasks = this.getAllTasks();
-      allTasks = [...allTasks, task];
-      this._writeToFile(TASKS_FILE_NAME, allTasks);
+   async deleteTask(id) {
+      await Task.destroy({
+         where: { id: id },
+      });
    }
 
-   deleteTask(index) {
-      const allTasks = this.getAllTasks();
-      allTasks.splice(index, 1);
-      this._writeToFile(TASKS_FILE_NAME, allTasks);
+   clearTasks = async () =>
+      await Task.destroy({
+         where: {},
+         truncate: true,
+      });
+
+   async sortTasks(direction) {
+      direction = direction === "down" ? "ASC" : "DESC";
+
+      return await Task.findAll({
+         order: [
+            ["text", direction],
+            ["id", "ASC"],
+         ],
+      });
    }
 
-   clearTasks = () => this._writeToFile(TASKS_FILE_NAME, []);
+   async checkMarkTask(isChecked, taskID) {
+      Task.findOne({ where: { id: taskID } }).then((task) => {
+         if (!task) {
+            throw new Error("No task found");
+         }
+         task.update({ status: isChecked });
+         task.update({ done: isChecked ? Date.now() : null });
+      });
+   }
 
    async getFilePokemonNames() {
       if (fs.existsSync(POKEMON_FILE_NAME)) {
```

Comment:
I would keep this method in a separate  fileManager class as it's not part of the DB logic

Reply:
Right, I will move those 2 methods to another file. Thanks!